### ISSUE TITLE
aws service discvery

### ISSUE THREAD

**Issue by yossi-r** [F5Networks/f5-appsvcs-extension#51]
##### ISSUE TYPE
 - Bug Report
 - Documentation Report

##### AS3 BUILD/ VERSION
3.7
```

```

##### BIGIP VERSION
13.1.1
```

```

##### SUMMARY

```
service discovery gets stuck and stops working. 
```

##### STEPS TO REPRODUCE
when using the example posted here:
https://clouddocs.f5.com/products/extensions/f5-appsvcs-extension/latest/declarations/discovery.html

the example for AWS service discovery has both static members and is using the service discovery. 
when posting this declaration it works, meaning i see both the static members and the ones discovered by the service discovery. 
then, if i delete the static ones from the declaration, it doesn't populate any members. 
even if i post the same declaration now with the static members it won't populate the dynamic ones. 
i can fix it by completely removing the pool from the declaration and posting it. then send another declaration with just the service discovery.

```
example of first declaration (with both static and service discovery)
{
  "class": "ADC",
  "schemaVersion": "3.7.0",
  "id": "5489432",
  "label": "ASM_policy_external_URL",
  "remark": "ASM_policy_external_URL",
  "controls": {
    "class": "Controls",
    "trace": true,
    "logLevel": "debug"
  },
  "Sample_sec_02": {
    "class": "Tenant",
    "HTTP_Service": {
      "class": "Application",
      "template": "http",
      "serviceMain": {
        "class": "Service_HTTP",
        "virtualAddresses": [
          "192.0.10.107"
        ],
        "snat": "auto",
        "pool": "web_pool",
        "policyWAF": {
          "use": "My_ASM_Policy"
        }
      },
        "web_pool": {
          "class": "Pool",
          "monitors": [
            "http"
          ],
          "members": [
            {
              "servicePort": 80,
              "addressDiscovery": "aws",
              "updateInterval": 1,
              "tagKey": "aws:autoscaling:groupName",
              "tagValue": "rosenbo-App10-master-app-appAutoscaleGroup-1HC0ALV8ZIFQ5",
              "addressRealm": "private",
              "region": "us-west-2"
            },
            {
              "enable": true,
              "servicePort": 443,
              "serverAddresses": [
                "192.0.2.7",
                "192.0.2.8"
              ]
            }
          ]
        },
      "My_ASM_Policy": {
        "class": "WAF_Policy",
        "url": "https://raw.githubusercontent.com/F5-use-cases/f5-rs-waf/master/roles/waf_policies/files/waf_policies/owasptop10-v01.xml",
        "ignoreChanges": true
      }
    }
  }
}
```
example of declaration with just the service discovery (works fine if used on a clean bigip, fails to populate members if used after the previous declaration) 
```
{
  "class": "ADC",
  "schemaVersion": "3.7.0",
  "id": "5489432",
  "label": "ASM_policy_external_URL",
  "remark": "ASM_policy_external_URL",
  "controls": {
    "class": "Controls",
    "trace": true,
    "logLevel": "debug"
  },
  "Sample_sec_02": {
    "class": "Tenant",
    "HTTP_Service": {
      "class": "Application",
      "template": "http",
      "serviceMain": {
        "class": "Service_HTTP",
        "virtualAddresses": [
          "192.0.10.107"
        ],
        "snat": "auto",
        "pool": "web_pool",
        "policyWAF": {
          "use": "My_ASM_Policy"
        }
      },
        "web_pool": {
          "class": "Pool",
          "monitors": [
            "http"
          ],
          "members": [
            {
              "servicePort": 80,
              "addressDiscovery": "aws",
              "updateInterval": 1,
              "tagKey": "aws:autoscaling:groupName",
              "tagValue": "rosenbo-App10-master-app-appAutoscaleGroup-1HC0ALV8ZIFQ5",
              "addressRealm": "private",
              "region": "us-west-2"
            }
          ]
        },
      "My_ASM_Policy": {
        "class": "WAF_Policy",
        "url": "https://raw.githubusercontent.com/F5-use-cases/f5-rs-waf/master/roles/waf_policies/files/waf_policies/owasptop10-v01.xml",
        "ignoreChanges": true
      }
    }
  }
}
```

##### EXPECTED RESULTS
discover members 

##### ACTUAL RESULTS
didn't discover members 

```

```

**Comment by dstokesf5** [comment]
This issue should be resolved by today's 3.8.0 release. Please try the new release to see if it fixes the problems you are encountering.

**Comment by kaustriaf5** [comment]
@yossi-r  closing this issue, as fix is available starting with 3.8 . Please re-open issue if this does not resolve the problem (or open a new one if you run into something else). Thanks!
